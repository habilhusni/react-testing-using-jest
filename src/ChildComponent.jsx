import React, { Component } from "React";

export default class ChildComponent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "husni"
    };
  }

  handleNotJonny = () => {
    this.setState({
      name: "habil"
    });
  };

  handleClick = () => {
    this.setState({
      name: this.state.name === "jonny" ? "husni" : "habil"
    });
  };

  componentDidMount() {
    this.setState({
      name: "jonny"
    });
  }

  render() {
    const { name } = this.state;
    return (
      <div>
        <div onClick={this.handleClick}>
          {name === "jonny" ? (
            <span>Statenya jadi jonny: {name}</span>
          ) : (
            <span>Statenya jadi bukan jonny: {name}</span>
          )}
        </div>
        <div>Ini div kedua</div>
      </div>
    );
  }
}
