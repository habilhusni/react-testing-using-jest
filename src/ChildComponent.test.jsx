import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";

import ChildComponent from "./ChildComponent";

Enzyme.configure({ adapter: new Adapter() });

it("Testing ChildComponent Snapshot", () => {
  const component = shallow(<ChildComponent />);
  expect(toJSON(component)).toMatchSnapshot();
});

it("Function handleClick Run Correctly", () => {
  const component = shallow(<ChildComponent />);
  component
    .find("div")
    .children()
    .first()
    .simulate("click");
  expect(component.state().name).toEqual("husni");
  component.instance().handleNotJonny();
  component.instance().handleClick();
  expect(component.state().name).toEqual("habil");
});

test("ComponentDidMount Run Correctly", () => {
  const component = shallow(<ChildComponent />);
  component.instance().componentDidMount();
  expect(component.state().name).toEqual("jonny");
});

describe("Conditional testing", () => {
  let component;

  beforeEach(() => {
    component = shallow(<ChildComponent />);
  });

  test("Condition if this.state.name === jonny", () => {
    component.instance().componentDidMount();
    expect(component.state().name).toEqual("jonny");
  });

  test("Condition if this.state.name !== jonny", () => {
    component
      .find("div")
      .children()
      .first()
      .simulate("click");
    expect(component.state().name).toEqual("husni");
  });
});

test("They have 2 Children div", () => {
  const component = shallow(<ChildComponent />);
  expect(component.find("div").children("div")).toHaveLength(2);
});
