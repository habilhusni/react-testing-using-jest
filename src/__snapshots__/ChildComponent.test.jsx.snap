// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ChildComponent Snapshot 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    <span>
      Statenya jadi jonny: 
      jonny
    </span>
  </div>
  <div>
    Ini div kedua
  </div>
</div>
`;
